integrations/tensorflow/python_projects/iree_tf/iree/tools/tf/scripts/iree_import_tf/__main__.py

```diff
@@ -64,25 +64,30 @@ def import_saved_model(
     *, output_path, saved_model_dir, exported_names, import_type, tags
 ):
     # From here there be dragons.
-    from tensorflow.python import pywrap_mlir
+    from tensorflow.mlir.experimental import (
+        convert_saved_model_to_mlir,
+        convert_saved_model_v1_to_mlir,
+        run_pass_pipeline,
+        write_bytecode,
+    )
 
     if import_type == "savedmodel_v2":
-        result = pywrap_mlir.experimental_convert_saved_model_to_mlir(
+        result = convert_saved_model_to_mlir(
             saved_model_dir, exported_names=exported_names, show_debug_info=False
         )
     elif import_type == "savedmodel_v1":
         # You saw it here, folks: The TF team just adds random positional params
         # without explanation or default. So we detect and default them on our
         # own. Because this is normal and fine.
-        sig = inspect.signature(pywrap_mlir.experimental_convert_saved_model_v1_to_mlir)
+        sig = inspect.signature(convert_saved_model_v1_to_mlir)
```

Comment:
Ah... I didn't realize how much this would break implicitly. We depend on a recently nightly release of TensorFlow, Colab lags a few versions behind, and installing nightly releases of TensorFlow is extremely brittle...

We have broken sample Colab notebooks: https://github.com/openxla/iree/actions/runs/5515709617 failing with `ImportError: cannot import name 'convert_saved_model_to_mlir' from 'tensorflow.mlir.experimental'`. I'm trying to install `tf-nightly` in Colab as a patch for _our_ samples, but user code will still be affected.

The detail I'm trying to figure out in Colab is how to get a compatible version of tf-nightly and numpy... I see errors relying on just tf-nightly's pip requirements:

```
!python -m pip install --upgrade tf-nightly
import tensorflow as tf

Numpy version error compiled with API 0x10 installed 0xf
```

and pip dependency resolver errors when I try to update numpy to work around that:

```
!python -m pip install --upgrade tf-nightly numpy>=1.24  # Needed for convert_saved_model_to_mlir in TF>=2.14

ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
numba 0.56.4 requires numpy<1.24,>=1.18, but you have numpy 1.25.1 which is incompatible.
tensorflow 2.12.0 requires numpy<1.24,>=1.22, but you have numpy 1.25.1 which is incompatible.
```

Reply:
Mmm, we could either roll back until next release or check if new function is defined and if not fall back to invoking the old. The latter should handle both cases and we can mostly be off the internal APIs. We can then delete it once next TF release. WDYT?

Reply:
> or check if new function is defined and if not fall back to invoking the old

I like that approach. I can also poke more at Colab to see if we can just change our samples to use tf-nightly there too. It would add extra overhead (time to install packages before the interesting bits) and complexity (could be unstable) though.

Reply:
I tested with `tf-nightly==2.14.0.dev20230705` and `iree-tools-tf==20230731.599`. But `iree-import-tf` still shows the error:

```
Traceback (most recent call last):
  File "/usr/local/google/home/cheyuw/iree/openxla-benchmark/test.venv/bin/iree-import-tf", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/usr/local/google/home/cheyuw/iree/openxla-benchmark/test.venv/lib/python3.11/site-packages/iree/tools/tf/scripts/iree_import_tf/__main__.py", line 54, in main
    import_saved_model(
  File "/usr/local/google/home/cheyuw/iree/openxla-benchmark/test.venv/lib/python3.11/site-packages/iree/tools/tf/scripts/iree_import_tf/__main__.py", line 67, in import_saved_model
    from tensorflow.mlir.experimental import (
ImportError: cannot import name 'convert_saved_model_to_mlir' from 'tensorflow.mlir.experimental' (/usr/local/google/home/cheyuw/iree/openxla-benchmark/test.venv/lib/python3.11/site-packages/tensorflow/_api/v2/mlir/experimental/__init__.py)
```

I'm wondering if it's because the method is called `convert_saved_model` instead of `convert_saved_model_to_mlir` (https://github.com/tensorflow/tensorflow/blob/b581c9f211b84c969adb6f5631eb2b3f964af818/tensorflow/python/compiler/mlir/mlir.py#L95)

Reply:
I sent a PR to update that, I forgot if it landed.